src/frontend/src/icp/utils/icrc.utils.ts

```diff
@@ -58,7 +59,7 @@ export const mapIcrcToken = ({
 
 	return {
 		id: parseTokenId(symbol),
-		network: ICP_NETWORK,
+		network: isTokenIcrcTestnet({ ledgerCanisterId }) ? ICP_PSEUDO_TESTNET_NETWORK : ICP_NETWORK,
```

Comment:
feel like this would be more readable with a dedicated function

```
network: mapICNetwork // + JSDOC
```

Or something